plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/launch/CLIParser.java

```diff
@@ -0,0 +1,108 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.launch;
+
+import java.io.File;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.eclipse.glsp.server.utils.LaunchUtil;
+
+public final class CLIParser {
+   private static final Logger LOG = Logger.getLogger(CLIParser.class);
+   private final CommandLine cmd;
+   private final Options options;
+   private final String processName;
+
+   public CLIParser(final String[] args, final String processName) throws ParseException {
+      this(args, getDefaultCLIOptions(), processName);
+   }
+
+   public CLIParser(final String[] args, final Options options, final String processName) throws ParseException {
+      this.cmd = new DefaultParser().parse(options, args);
+      this.options = options;
+      this.processName = processName;
+   }
+
+   public boolean optionExists(final String identifier) {
```

Comment:
Just out of curiosity: Is there a reason you chose 'optionExists' over 'hasOption'? In my opinion 'hasOption' is both shorter and more common.